model/dependencies.go

```diff
@@ -102,20 +108,39 @@ func (di *dependencyIncluder) expandDependencies(pair TVPair, depends []TaskUnit
 		case d.Variant == AllVariants && d.Name == AllDependencies: // task = *, variant = *
 			// Here we get all variants and tasks (excluding the current task)
 			// and add them to the list of tasks and variants.
-			for _, v := range di.Project.FindAllVariants() {
-				for _, t := range di.Project.FindTasksForVariant(v) {
-					if !(t == pair.TaskName && v == pair.Variant) {
-						deps = append(deps, TVPair{TaskName: t, Variant: v})
+			for _, v := range di.Project.BuildVariants {
+				for _, t := range v.Tasks {
+					projectTask := di.Project.FindTaskForVariant(t.Name, v.Name)
+					if projectTask != nil {
+						if projectTask.SkipOnRequester(di.requester) {
+							continue
+						}
+						if t.Name == pair.TaskName && v.Name == pair.Variant {
```

Comment:
This if case doesn't actually rely on getting projectTask

Reply:
The check above it does

Reply:
I think I just meant this if statement could be moved out of the block (i.e. 120) so it can still be checked even if projectTask does == nil (although if projectTask == nil should we really be adding a dependency?)

Reply:
got it, fixed